### DISCUSSION TITLE
PlanetScale Read Only Replicas

### DISCUSSION THREAD

**Discussion by enqqi** [planetscale/discussion#486] [Q&A] [answered]
Does anyone knows if PlanetScale auto routes requests made with the [serverless driver](https://github.com/planetscale/database-js) to the replica that is near to the user/server? 

I appreciate any clarification on how read only replicas work or if I have to setup a different connection for each one.

Thanks!

Edit: I am using Cloudflare workers

**Comment by jgreet** [comment, marked as answer]
There's no automatic routing of requests to read-only replicas. You'll have to create separate credentials for reach read-only replica and manage connecting to them in your application.

See also https://github.com/planetscale/discussion/discussions/400

**Reply by toman** [reply]
I don't believe that's right. Other similar questions have the answer that using
`USE @replica`
at the beginning of a connection will send the subsequent SELECTs to the read only replica. This is handled at the load balancer. This is mentioned here https://planetscale.com/docs/concepts/replicas as well.

**Reply by jgreet** [reply]
`USE @replica` will work for the standby replicas in any production branch. The primary and these replicas are all in the same AWS or GCP region.

To query additional "Read Only Regions" created in other locations, which is what I believe the original question was about, you need to use credentials specific to each individual read-only region. There's currently no routing of requests to the nearest read-only region.

https://planetscale.com/docs/concepts/read-only-regions
